src/toolbars/TimeToolBar.h

```diff
@@ -65,6 +67,7 @@ class TimeToolBar final : public ToolBar
    static const int maxDigitH = 100;
 
    Observer::Subscription mSubscription;
+   Observer::Subscription mTimeSignatureChangedSubscription;
```

Comment:
TimeToolBar and SelectionBar listen for changes and then inform their NumericTextCtrls.  This feels unsatisfactory -- the controls should do their own subscriptions.  But getting AudacityProject& into the constructors of all NumericTextCtrl everywhere is too bothersome.

I believe these toolbars (and the spectral selection bar, never mind it) are the only non-modal text controls.

There are others in TimeDialog and the label editing dialog and effects that are modal.

Edge case (I haven't tried it yet):  New project.  Set time signature, and make beat the preferred format.  Another new project.  Set a different time signature.  Switch back to first project.  Open a modal NumericTextCtrl.

Which time signature is used?

If it's the wrong one, it's because the NumericTextCtrl (and NumericConverter) are not correctly initialized for the project's time signature but instead use the last values saved in preferences.

Please just note this, don't try to fix it!

Reply:
Yeah, just wait me to press push xD

Reply:
> Edge case (I haven't tried it yet): New project. Set time signature, and make beat the preferred format. Another new project. Set a different time signature. Switch back to first project. Open a modal NumericTextCtrl.
> 
> Which time signature is used?
> 
> If it's the wrong one,

There are many rough-edges with having multiple projects open - this is just the lates one.

Reply:
Oh damn it's outdated now

Reply:
This is a problem with commit by commit review, yes. Especially when every commit does a step by step transformation.

I hope you understand my concerns about matryoshka PRs now.

Reply:
@petersampsonaudacity have You tried this edge case? The latest commit as of this comment may include fixes for that